python/ray/dag/collective_node.py

```diff
@@ -124,8 +134,21 @@ def execute(self, send_buf: "torch.Tensor") -> "torch.Tensor":
         if not isinstance(send_buf, torch.Tensor):
             raise ValueError("Expected a torch tensor")
         communicator = self.get_communicator()
-        recv_buf = torch.empty_like(send_buf)
-        communicator.allreduce(send_buf, recv_buf, self._op)
+        if self.comm_op == self.ALLREDUCE:
+            recv_buf = torch.empty_like(send_buf)
+            communicator.allreduce(send_buf, recv_buf, self._op)
+        elif self.comm_op == self.REDUCESCATTER:
+            world_size = len(self._actor_handles)
+            assert (
```

Comment:
I'm planning on adding a test on throwing an error when input tensors' first dimension is not divisible by the number of reduce scatter participants. This is a `RayTaskError`, I'm not sure what is the expected behavior of compiled graph when this error is generated.

Reply:
Sounds good. Raise ValueError for now.